features/telemetry/protocols/collection/src/main/java/org/opennms/netmgt/telemetry/protocols/collection/AbstractPersistingAdapter.java

```diff
@@ -77,6 +81,17 @@
     @Autowired
     private PersisterFactory persisterFactory;
 
+    @Autowired
+    private ThresholdingService thresholdingService;
+
+    // Changed to False if no ThresholdingService has been wired.
+    private boolean isThresholdingEnabled = true;
```

Comment:
May be worth marking this as `volatile` or using an `AtomicBoolean` since the value can be changed by different threads (`handleMessageLog` should be thread-safe).

Reply:
changed to `AtomicBoolean `.